OptionalExtensions/Source/OptionalExtensions.swift

```diff
@@ -45,4 +45,27 @@ public extension Optional {
     var isNone: Bool {
         return !isSome
     }
+
+    func flatten() -> Any? {
+        return self.unwrap()
+    }
+}
+
+protocol Unwrappable {
+
+    func unwrap() -> Any?
+}
```

Comment:
Since it does flatten recursively, the final unwrapped value's type is not `Wrapped`. For example, `let number: Int??? = 3` has `Optional` type with its `Wrapped` to be another `Optional` while the final unwrapped value's type is Int.